mindsdb_native/libs/data_sources/gcs_ds.py

```diff
@@ -0,0 +1,39 @@
+import os 
+
+from google.cloud import storage
+
+from mindsdb_native.libs.data_types.data_source import DataSource
+from mindsdb_native.libs.data_sources.file_ds import FileDS
+from mindsdb_native import F
+
+
+class GCSDS(DataSource):
+
+    def _setup(self, bucket_name, file_path, path_to_auth_json):
+
+        self._bucket_name = bucket_name
+        self._file_name = os.path.basename(file_path)
+
+        gc_client = storage.Client.from_service_account_json(json_credentials_path=path_to_auth_json)
```

Comment:
Maybe allow `path_to_auth_json` to be either a  path or a json and rename it to `auth_json`, in which case you can use `json.loads(auth_json)` to read the credentials if it's a valid json and if it isn't than assume it's a file and try reading it from the file system.

Reply:
I added both the options to load from a file through path as well as json itself. 
PTAL! and let me know if I should change any more.